src/estimators/exact_set.py

```diff
@@ -81,10 +98,13 @@ def __init__(self):
     EstimatorBase.__init__(self)
 
   def __call__(self, sketch_list):
-    """Return len(sketch) - 1."""
+    """Returns frequency histogram of sketch with total cardinality reduced by one."""
     e = LosslessEstimator()
     histogram = e(sketch_list)
-    return [histogram[0]-1] + [histogram[1:]]
+    if sum(histogram) == 0:
+      raise ValueError("Attempt to create a histogram with a negative value!")
+    histogram = [min(h, histogram[0]-1) for h in histogram]
```

Comment:
I don't understand what min(h, histogram[0]-1) is doing. I don't see how this gets you what your docstring is promising anyway.

It's possible that this "LessOneEstimator" does not make much sense in the frequency case.

Also, you don't really need the temporary variable.

Reply:
I've added an explanation of what is happening.